Initial commit

joefitter opened the pull request:
This component is a hof-behaviour, designed to be mixed into `hof-form-controller` or an extension. It extends functionality by running hooks before and after pipeline methods. This can be used to extend form/field functionality within the main controller pipeline.

joefitter committed:
Initial commit

This component is a hof-behaviour, designed to be mixed into `hof-form-controller` or an extension. It extends functionality by running hooks before and after pipeline methods. This can be used to extend form/field functionality within the main controller pipeline.

joefitter committed:
added travis config

joefitter committed:
removed sinomocha as fails peer dependency

joefitter committed:
use npm@3

joefitter committed:
clear travis cache

lennym reviewed (changes requested):
I'm a little concerned that the tests do a bit too much of testing *how* the thing works - that is, they assert on the existence and behaviour of `runHooks` rather than on the overall effect of executing behaviour in pre and post method bindings.

Ideally `runHooks` should be completely invisible externally, and therefore shouldn't be tested directly.

With your permission I'll do a little rewrite of the tests.

lennym committed:
Move index file content into ./lib

IMO, keeping source code in a lib directory makes it easier to find within a project when the project root contains a lot of different config files and dotfiles, which can make spotting the `index.js` file tricky.

lennym committed:
Refactor tests to test effects not implementation

The previous iteration of the tests tested the internals of the behaviour as opposed to its effect when applied to a controller.

Replace these tests with test which apply the behaviour to a controller and then assert that the controller now works in the expected way.

lennym committed:
Fix linting errors

Disables no-underscore-dangle rule because dangling underscores are used widely in this project on account of inheriting methods from hof-form-controller.

lennym committed:
Merge pull request #2 from UKHomeOfficeForms/init-test-refactor

Refactor tests to tests effects and not implementation

lennym commented:
![](http://i.imgur.com/bT7xJnA.gif)
